src/nox_uv/__init__.py

```diff
@@ -77,17 +77,15 @@ def session(
     @functools.wraps(function)
     def wrapper(s: nox.Session, *_args: Any, **_kwargs: Any) -> None:
         if s.venv_backend == "uv":
-            env: dict[str, Any] = {"UV_PROJECT_ENVIRONMENT": s.virtualenv.location}
+            s.env["UV_PROJECT_ENVIRONMENT"] = s.virtualenv.location
 
             # UV called from Nox does not respect the Python version set in the Nox session.
             # We need to pass the Python version to UV explicitly.
             if s.python is not None:
-                env["UV_PYTHON"] = s.python
+                s.env["UV_PYTHON"] = s.python
```

Comment:
not sure what to do about the typing here because the type checker isn't seeing that we effectively changed the type by using the:

```python
if s.python is not None:
    s.env["UV_PYTHON"] = s.python
```

changes the type of `s.python` to effectively `str` instead of `str | None`.

Reply:
nevermind. if I wrap `s.python` to `str(s.python)` the type checker is happy.